wowup-electron/src/app/pages/my-addons/my-addons.component.ts

```diff
@@ -787,6 +803,31 @@ export class MyAddonsComponent implements OnInit, OnDestroy, AfterViewInit {
           row.addon.isIgnored = false;
         }
 
+        row.addon.autoUpdateNotificationsEnabled = isAutoUpdate;
+
+        this.addonService.saveAddon(row.addon);
+      }
+
+      this.rowData = rows;
+    } catch (e) {
+      console.error(e);
+      this._operationErrorSrc.next(e);
+    }
+  }
+
+  public onClickAutoUpdateAddonsNotifications(listItems: AddonViewModel[]): void {
+    const isAutoUpdateNofications = _.every(listItems, (listItem) => listItem.addon?.autoUpdateNotificationsEnabled === false);
+    const rows = [...this._baseRowData];
+    try {
+      for (const listItem of listItems) {
+        const row = _.find(rows, (r) => r.addon?.id === listItem.addon?.id);
+        if (!row || !row.addon) {
+          console.warn("Invalid row data");
```

Comment:
If the idea is to keep the warning here, can you add some specific information that we can use when someone shares their logs? Perhaps the addon id or something else to uniquely distinguish the row.

What if both `r.addon?.id` and `listItem.addon?.id` are null, will it equate to `null === null`? I'm not too familiar with how this is handled in typescript. My guess is that `row.addon` is false-y here because of `undefined` and it works?

Reply:
Not certain myself, but from what I think I know I would assume that it would indeed equate to a `null === null` check. The `onClickUpdateAddonsNotifications` method is based on the `onClickAutoUpdateAddons` given how similar their functionality are, so the null checking and the conditions for when to log are exactly the same.

I did push a minor adjustment to the warning though, expanding it with the `listItem.addon?.id` which should either provide the logs with the problematic id or simply let it know that it somehow was null